client/src/java/com/zimbra/client/ZSearchParams.java

```diff
@@ -488,41 +502,53 @@ public void setQueryString(String value) {
 
     @Override
     public Set<MailItemType> getMailItemTypes() {
-        throw new UnsupportedOperationException("ZSearchParams method not supported yet");
+        Set<MailItemType> result = EnumSet.noneOf(MailItemType.class);
+        for (String token : Splitter.on(',').trimResults().split(mTypes)) {
+            MailItemType type = MailItemType.valueOf(token.toUpperCase());
+            if (type != MailItemType.UNKNOWN) {
+                result.add(type);
+            } else {
+                throw new IllegalArgumentException(token);
+            }
+        }
+        return result;
     }
 
     @Override
     public ZimbraSearchParams setMailItemTypes(Set<MailItemType> values) {
-        throw new UnsupportedOperationException("ZSearchParams method not supported yet");
+        mTypes = Joiner.on(",").join(values);
+        return this;
     }
 
     @Override
     public ZimbraSortBy getZimbraSortBy() {
-        throw new UnsupportedOperationException("ZSearchParams method not supported yet");
+        return mSortBy.toZimbraSortBy();
```

Comment:
There is a potential NPE here, because SearchSortBy.fromZimbraSortBy can return null.